oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -148,6 +158,10 @@
     this.serviceAccountUser = serviceAccountUser;
     this.projectId = projectId;
     this.quotaProjectId = quotaProjectId;
+    if (lifetime > TWELVE_HOURS_IN_SECONDS) {
+      throw new IllegalStateException(LIFETIME_EXCEEDED_ERROR);
+    }
+    this.lifetime = lifetime == 0 ? DEFAULT_LIFETIME_IN_SECONDS : lifetime;
```

Comment:
With the builder setting the default already, do we still want this check for 0?

Reply:
Just in case the user calls `setLifetime(0)` in builder and overrides the default lifetime to 0.

Reply:
now moved `lifetime == 0 ? DEFAULT_LIFETIME_IN_SECONDS : lifetime` logic to the builder's `setLifeTime` instead